src/modals/SendModal/index.tsx

```diff
@@ -59,40 +74,59 @@ const SendModal = ({ onClose }: SendModalProps) => {
     currentNetwork
   } = useGlobalContext()
   const { setAddress } = useAddressesContext()
+  const { dappTransactionData, requestEvent, walletConnect } = useWalletConnectContext()
   const [title, setTitle] = useState('')
-  const [transactionData, setTransactionData] = useState<SendTransactionData | undefined>()
+  const [transactionData, setTransactionData] = useState<SendTransactionData | undefined>(dappTransactionData)
   const [isLoading, setIsLoading] = useState(false)
-  const [step, setStep] = useState<StepIndex>(1)
+  const [step, setStep] = useState<StepIndex>(StepIndex.Send)
   const [isConsolidateUTXOsModalVisible, setIsConsolidateUTXOsModalVisible] = useState(false)
   const [consolidationRequired, setConsolidationRequired] = useState(false)
   const [isSweeping, setIsSweeping] = useState(false)
   const [sweepUnsignedTxs, setSweepUnsignedTxs] = useState<SweepAddressTransaction[]>([])
   const [unsignedTxId, setUnsignedTxId] = useState('')
   const [unsignedTransaction, setUnsignedTransaction] = useState('')
+  const [contractAddress, setContractAddress] = useState('')
   const [fees, setFees] = useState<bigint>()
   const theme = useTheme()
 
   useEffect(() => {
-    if (step === 1) {
-      setTitle('Send')
-    } else if (step === 2) {
-      setTitle('Info Check')
-    } else if (step === 3) {
-      setTitle('Password Check')
-    }
+    setTitle(stepToTitle.get(step) || 'Unknown')
```

Comment:
Are you sure this works fine when going back and forth the steps (pressing the back button)? `'Unknown'` should never happen, we don't want to display that.

Reply:
It should never happen but the TypeScript type checker was not happy (in all fairness) because `.get()` can be returned undefined if the key is not found. I don't think there's a way to avoid this? Even with a POJO, it's possible to get `undefined`... But maybe TypeScript will allow me to be strict in that the object will never have it, i.e. `const stepToTitle: StepToTitle = { [Step.Send]: 'Send', ... }; type StepToTitle = { [k: Step]: string };` might not work though because `k` must be a `string` or `Symbol`